back-end/www/models/model_operations/answer_operations.py

```diff
@@ -0,0 +1,306 @@
+        raise Exception("The photo year shall be an integer")
+
+    answer = Answer(user_id=user_id, location_id=location_id, year_old=year_old,
+            year_new=year_new, source_url_root=source_url_root, land_usage=land_usage,
+            expansion=expansion, is_gold_standard=is_gold_standard,
+            bbox_left_top_lat=bbox_left_top_lat, bbox_left_top_lng=bbox_left_top_lng,
+            box_bottom_right_lat=bbox_bottom_right_lat,
+            bbox_bottom_right_lng=bbox_bottom_right_lng, zoom_level=zoom_level)
+
+    db.session.add(answer)
+    db.session.commit()
+
+    return answer
+
+
+def create_answer_with_result(user_id, location_id, year_old, year_new,
+        source_url_root, land_usage, expansion, is_gold_standard,
+        bbox_left_top_lat, bbox_left_top_lng, bbox_bottom_right_lat,
+        bbox_bottom_right_lng, zoom_level):
+    """
+    Create an answer and return if it passed or not by checking the gold standards.
+
+    Parameters
+    ----------
+    user_id : int
+        Foreign key to the user table.
+    location_id : int
+        Foreign key to the location table.
+    year_old: int
+        The year where the satellite photo was taken from our geo sources.
+    year_new: int
+        A newer photo to be compared with the one taken in year_old.
+    source_url_root : str
+        URL to store the location on the map.
+    land_usage : int
+        User's answer of judging if the land is a farm or has buildings.
+        (check the answer table in model.py for the meaning of the values)
+    expansion : int
+        User's answer of judging the construction is expanded.
+        (check the answer table in model.py for the meaning of the values)
+    is_gold_standard : bool
+        If it's a golden standard answer provided by admnin.
+    bbox_left_top_lat : float
+        The latitude of the top-left corner of the bounding box for displaying the focus.
+    bbox_left_top_lng : float
+        The longitude of the top-left corner of the bounding box for displaying the focus.
+    bbox_bottom_right_lat : float
+        The latitude of the bottom-right corner of the bounding box for displaying the focus.
+    bbox_bottom_right_lng : float
+        The longitude of the bottom-right corner of the bounding box for displaying the focus.
+    zoom_level : int
+        The zoom level for displaying the location.
+
+    Returns
+    -------
+    compare_result : int
+        Compare with the gold answer.
+        0 means Pass
+        1 means Fail
+        -1 means that the gold standard does not exist
+
+    Raises
+    ------
+    exception : Exception
+        Type checking for land_usage, expansion, year_old, year_new, and is_gold_standard.
+    """
+    new_answer = create_answer(user_id, location_id, year_old, year_new,
+            source_url_root, land_usage, expansion, is_gold_standard,
+            bbox_left_top_lat, bbox_left_top_lng, bbox_bottom_right_lat,
+            bbox_bottom_right_lng, zoom_level)
```

Comment:
Why creating the answer and put it into the database first? I thought we should only create the answer after we make sure that the answer passes the data quality check (using the gold standard answer)?